lib/metababel/bt2_values_generator.rb

```diff
@@ -125,7 +125,7 @@ class BTValueCLass::IntegerUnsigned < BTValueCLass::Scalar
 
     def initialize(name, usr_default_value)
       bt_type = self.class.instance_variable_get(:@bt_type)
-      if !usr_default_value.nil? and (!usr_default_value.is_a? Integer or !usr_default_value.between?(0, (2**64) - 1))
+      if !usr_default_value.nil? && (!usr_default_value.is_a?(Integer) || !usr_default_value.between?(0, (2**64) - 1))
```

Comment:
Great, here there is a substantial difference, since `&&` have higher precedence than `||` while `and` and `or` have the same precedence.

Reply:
I don't understand. The code should be the same, thanks to the parenthesis.

Reply:
Yoy are right, I did not saw them !!!